sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -404,6 +435,70 @@ class FilesElement extends Polymer.Element {
       return Promise.resolve(null);
     }
   }
+
+  /**
+   * Creates a directory picker modal to get the user to choose a destination for the
+   * selected item, sends a copy item API call, then refreshes the file list. This only
+   * works if exactly one item is selected.
+   * TODO: Consider allowing multiple items to be copied.
+   */
+  _copySelectedItem() {
+    const selectedIndices = this.$.files.getSelectedIndices();
+
+    if (selectedIndices.length === 1) {
+      const i = selectedIndices[0];
+      const selectedObject = this._fileList[i];
+
+      const options: DialogOptions = {
+        title: 'Copy Item',
+        okLabel: 'Copy Here',
+        big: true,
+      };
+      Utils.showDialog(DirectoryPickerDialogElement, options)
```

Comment:
Should this function and the rename function return a Promise to make testing easier?